notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
More on topic for this PR, it seems like these should still be computed during scrape time (at least the queue length, the capacity will always be the same), so the collector interface should be kept. Or alternatively you could use a `GaugeFunc` which has access to the specific notifier so that it has access to its queue to compute the length.